packages/edit-site/src/components/use-edited-entity-record/index.js

```diff
@@ -26,7 +27,11 @@ export default function useEditedEntityRecord( postType, postId ) {
 				usedPostType,
 				usedPostId
 			);
-			const _isLoaded = !! usedPostId;
+			const _isLoaded = hasFinishedResolution( 'getEditedEntityRecord', [
+				'postType',
+				usedPostType,
+				usedPostId,
+			] );
```

Comment:
We should probably check if `usedPostId` is defined. Otherwise, you'll see the `null` flash in the title when visiting `wp-admin/site-editor.php`, because the editor is resolving the edited template from the server, and we get a false positive.

```js
const _isLoaded =
				usedPostId &&
				hasFinishedResolution( 'getEditedEntityRecord', [
					'postType',
					usedPostType,
					usedPostId,
				] );
```

Reply:
Ah, of course, good catch! Added in d032f4dd8915a81659c9145e204f8133b03da853

Reply:
I don't understand why the extra `usedPostId && ...` check was needed here. If `usedPostId` is `null`, the `hasFinishedSelector` returns `true` for the `null` value?

Reply:
I'd say the idea is to not call the `hasFinishedResolution` selector at all if the post ID is `null` since we already know the result.

Reply:
There's a point where both selectors use `undefined` as `postId`. The `getEditedEntityRecord` will resolve and return an empty object, and `hasFinishedResolution` will falsy report that our template was loaded.

Example:

```js
const { testRecord, testResolved } = useSelect( ( select ) => {
	const { getEditedEntityRecord, hasFinishedResolution } =
		select( coreStore );

	return {
		testRecord: getEditedEntityRecord(
			'postType',
			'wp_template',
			undefined
		),
		testResolved: hasFinishedResolution( 'getEditedEntityRecord', [
			'postType',
			'wp_template',
			undefined,
		] ),
	};
} );

console.log( { testRecord, testResolved } )
```

Reply:
Oh, now I see what's happening. A `getEntityRecord( 'postType', 'wp_template', undefined )` call doesn't make sense. It will do a lookup like `return items[ key ]`, but there is never a template with `id === undefined`, so it will always return `undefined`.

But it still calls the resolver, where `undefined` key _does_ make sense. Instead of fetching a specific template from `/templates/{key}`, it will fetch all templates from `/templates` and store the list.

That's why we end up in a situation where `getEntityRecord` returns `undefined` (there's no item with `undefined` key) and `hasFinishedResolution` returns `true` (the list is loaded).

If `usedPostId` is not known, the hook shouldn't call any selector at all and return an empty-ish object right away.